### PR TITLE
Cron workflow: APIs

### PR DESCRIPTION
**Pull request by sdminonne**
APIs (very very similar to CronJob) and generated code

@clamoriniere1A @EricMountain-1A 

Thanks to having a look

### CODE REVIEW

File: docs/devel.md [hunk 1]
```diff
@@ -33,3 +33,16 @@ now the informers:
 ```shell
  ${GOPATH}/bin/informer-gen --input-dirs "github.com/amadeusitgroup/workflow-controller/pkg/api/workflow/v1" --versioned-clientset-package "github.com/amadeusitgroup/workflow-controller/pkg/client/versioned" --listers-package "github.com/amadeusitgroup/workflow-controller/pkg/client/listers" --output-package "github.com/amadeusitgroup/workflow-controller/pkg/client/informers"   -h "hack/boilerplate.go.txt"
 ```
+
+
+# How to generate api for cron workflows
+
+```shell
+$ go get k8s.io/code-generator/...
```
Comment (clamoriniere1A) [inline docs/devel.md]: the problem with this method is that you don't used a specific version the code-generator. And so someone who will try to regenerate the files, may not have the same result.
Personally, I put the `code-generator` as glide dependencies.

File: hack/update-codegen.sh [hunk 2]
```diff
@@ -4,13 +4,13 @@ set -o nounset
 set -o pipefail
 
 #deep copy
-$GOPATH/bin/deepcopy-gen --input-dirs "github.com/amadeusitgroup/workflow-controller/pkg/api/workflow/v1" \
+$GOPATH/bin/deepcopy-gen --input-dirs "github.com/amadeusitgroup/workflow-controller/pkg/api/workflow/v1,github.com/amadeusitgroup/workflow-controller/pkg/api/cronworkflow/v1" \
                          -p github.com/amadeusitgroup/workflow-controller/pkg/api/\
                          --bounding-dirs github.com/amadeusitgroup/workflow-controller/pkg/api \
                          -h "hack/boilerplate.go.txt"
```
Comment (clamoriniere1A) [inline hack/update-codegen.sh]: I understood why the generated files didn't contain the license: `boilerplate.go.txt` didn't contain it :)

File: docs/devel.md (hunk 1 above)
Comment (sdminonne) [reply docs/devel.md]: I know and it's in my TODO list

File: hack/update-codegen.sh (hunk 2 above)
Comment (sdminonne) [reply hack/update-codegen.sh]: :1st_place_medal: mystery solved then

File: docs/devel.md (hunk 1 above)
Comment (clamoriniere1A) [reply docs/devel.md]:
```yaml
- package: k8s.io/code-generator
  version: kubernetes-1.9.1
```

**Comment by sdminonne** [comment]
This is simply a rebase. Merging on @clamoriniere1A LGTM

**Pull request merged**